sql/plan/foreign_key_check.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+//TODO: doc and check field names
+type ForeignKeyReferenceChecker struct {
```

Comment:
Is this node going to be responsible to checking parent references and cascading to children? or will those be separate nodes?

Reply:
They’re going to be separate nodes at first, and then if I see a large common path between the two I’ll merge them, I think that’s the easiest developmental approach.